### PR TITLE
Style Guide for Help Text

### PR DESCRIPTION
**Pull request by knoll-alex**
Hello! I created a Style Guide for future iterations of the user-facing help text in LORIS. This is a set of guidelines to follow for consistency.

#### Testing instructions

1. Review for content. Comment as necessary. 
2. Ensure image is appearing.

Thanks for welcoming me on the LORIS team for a brief time!

### CODE REVIEW

File: docs/HelpStyleGuide.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  * i.e. in the Metadata column of the table
+  * i.e. in the Imaging menu
+* **Bold** when referring to any clickable button, including modules and tabs
+  * DON’T call it a button
+    * i.e. Click **Submit** (rather than “click the **Submit** button”)
+  * DO call it a module or tab
+    * i.e. In the **Imaging Browser** module
+    * i.e. Click the **Upload** tab
+  * You can also bold when defining features in a list, to separate the term being defined from the definition itself, i.e. when describing columns in a table:
+     **Column heading 1**: description
```
Comment (christinerogers) [inline docs/HelpStyleGuide.md]: These aren't rendering on their own lines.  Take a look, @knoll-alex ?

Comment (knoll-alex) [reply docs/HelpStyleGuide.md]: thanks! this should be fixed.

File: docs/HelpStyleGuide.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  * i.e. “This module allows *you* to browse…” (rather than “This module allows *users* to browse…”)
+* Keep it concise. Avoid unnecessary information.
+  * Try not to be repetitive. 
+  * i.e. the Help text for the Imaging Browser module begins with the title Imaging Browser. You don’t need to start text with “In the Imaging Browser module...”, because the user already knows the module based on the title. Instead, keep it short and say “In this module…”
+* Assume the user is tech-savvy enough in the year 2020 to know about features that are intuitive or common across modern platforms, so you don’t need to repeat in every Help text
+  * i.e. clicking the column headers to re-sort a table
+  * i.e. clicking a Download button to download something
+  * i.e. navigate pages of a table using the arrows provided
+* Use [Plain Language](https://www.plainlanguage.gov/resources/articles/elements-of-plain-language/)
+  * As a guideline, refer to [this list](https://www.plainlanguage.gov/guidelines/words/use-simple-words-phrases/) of words to avoid, and alternate suggestions
```
Comment (PapillonMcGill) [inline docs/HelpStyleGuide.md]: I don't like that we refer to foreign government rules. Not only theses can change without us noticing but they may not be the best in our situation.

File: docs/HelpStyleGuide.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    * i.e. In the **Imaging Browser** module
+    * i.e. Click the **Upload** tab
+  * You can also bold when defining features in a list, to separate the term being defined from the definition itself, i.e. when describing columns in a table:<br>
+     **Column heading 1**: description<br>
+     **Column heading 2**: description<br>
+     Etc.
+     
+## Tone
+
+Yes, LORIS is used for high-level research and yes, the people using LORIS are very smart and yes, we want people in the scientific community to take LORIS seriously. At the same time, the easier any user finds a text to read, the more they’ll retain and use the information. And a friendly tone will help them understand that LORIS is built and maintained by humans. There’s a lot of research that shows people will engage and have more patience with a piece of software when they’re reminded that there’s more than just a computer behind it. Using a more conversational tone will subtly have this effect. We’re not talking about humour and exclamation marks—just small tactics that make content as easy to understand as possible.
```
Comment (PapillonMcGill) [inline docs/HelpStyleGuide.md]: replace lines 48-50 with 
`LORIS is used by scientific community worldwide, this include countries where English is a second or even third language. Thus, it is important that LORIS documentation (instruction, help text, ...) be written in a way comprehensible by people with limited knowledge and understanding of English language and expression.`
or something more similar.

File: docs/HelpStyleGuide.md (hunk 2 above)
Comment (christinerogers) [reply docs/HelpStyleGuide.md]: @PapillonMcGill I think we could still include these links for context(inspiration), without saying that we are strictly following them -- would you be ok with that?

File: docs/HelpStyleGuide.md (hunk 3 above)
Comment (christinerogers) [reply docs/HelpStyleGuide.md]: great suggestions @PapillonMcGill,  and this could probably be shortened as you suggest.

Comment (driusan) [inline docs/HelpStyleGuide.md]: Do you have any references for "a lot of research" (both out of curiousity, and in case someone asks in the future) that could be referenced here?

**Comment by knoll-alex** [comment]
Thank you @PapillonMcGill for your review and suggestions! I have updated the content to better reflect the tone you're looking for. I wasn't sure if you wanted me to remove line 50 as well - let me know and I can remove it if need be. I removed links to the government website, and instead added an article at the end which references research-based information on Plain Language instead.

**Comment by christinerogers** [comment]
Hi @knoll-alex This week the team is rebasing all of our PRs open on `major` to `master` -- 
If this is doable for you, please take a moment to hop this onto `master`, and if not, don't hesitate to let me know.  Thanks!

File: docs/HelpStyleGuide.md (hunk 3 above)
Comment (knoll-alex) [reply docs/HelpStyleGuide.md]: Hi @driusan - I removed this line (along with the rest of that paragraph), but still happy to provide some context to this claim. I added an article at the end that talks about Plain Language, and cites a usability study that was conducted in scientific fields. I hope this will suffice!

**Comment by PapillonMcGill** [comment]
Yes please, could you remove line 50?

File: docs/HelpStyleGuide.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    * i.e. In the **Imaging Browser** module
+    * i.e. Click the **Upload** tab
+  * You can also use bold when defining features in a list, to separate the term being defined from the definition itself, i.e. when describing columns in a table:<br>
+     **Column heading 1**: description<br>
+     **Column heading 2**: description<br>
+     Etc.
+     
+## Tone
+
+LORIS is used by scientific community worldwide, including countries where English is a second, or even third language. Therefore, it is important that LORIS documentation (help text, setup guides, etc.) be written in a way that it comprehensible by people with limited knowledge and understanding of English language and expression.
```
Comment (christinerogers) [inline docs/HelpStyleGuide.md]: To shorten, and fix 2 grammar issues: 
```suggestion
LORIS is used by scientific community members worldwide, which means it's possible that most users are non-native English speakers. Writing our Help text or any LORIS documentation (setup guides, etc.) in a way that it is quickly comprehensible by people with many different levels of English language and expression.
```

File: docs/HelpStyleGuide.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+Please note, this is NOT about “dumbing things down”. It’s simply about using the most direct language and tone to get the information across to the user in the least amount of time, with the least amount of friction. The quicker a user understands something, the happier they are, the more trust they have with the software, and the less silly questions you can expect to get to the dev mailing list!
```
Comment (christinerogers) [inline docs/HelpStyleGuide.md]: Paring this down just a little, after chatting with @PapillonMcGill 
```suggestion
This means using more direct language and tone to get information across to the user in the least amount of time, with the least amount of friction. The quicker a user understands something, the happier they are.  Easily understandable documentation also builds trust in the software, in addition to decreasing the amount of support questions we get.
```

File: docs/HelpStyleGuide.md (hunk 4 above)
Comment (PapillonMcGill) [reply docs/HelpStyleGuide.md]: `... in a way that is quickly ...`

I would just do a little change to the last sentence.

Comment (christinerogers) [reply docs/HelpStyleGuide.md]: Ok thanks - i updated my suggestion to include your change @PapillonMcGill

File: docs/HelpStyleGuide.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+## How to Store and Name Files
+
+Help text content is stored into the aces/Loris repo as markdown (.md) files within the help/subdirectory of the module it belongs to. For example, the Help text for Imaging Browser main page is located here: `modules/imaging_browser/help/imaging_browser.md`. 
+
+Some modules have more than one Help text file to correspond with subpages. These files exist in the same `help/` folder, but are named differently - they must match exactly the php/file name that spawns the page. For example, the Imaging Browser module has a subpage called “ViewSession”. Its help file is stored as: `modules/imaging_browser/help/viewsession.md`. 
+
+## Markdown
+
+Formatting should follow markdown guidelines, with some exceptions. This is because LORIS’s current markdown template may not cover all possible markdown features (i.e. bullets). Help text should not include HTML tags for security reasons. When creating a markdown file, use only the following styles, or else talk to the LORIS technical team about expanding our Markdown features:
```
Comment (johnsaigle) [inline docs/HelpStyleGuide.md]:
```suggestion
Formatting should follow markdown guidelines, with some exceptions. This is because LORIS’s current markdown template may not cover all possible markdown features (e.g. bullets). Help text cannot be formatted with HTML so it should not be included in help text. When creating a markdown file, use only the following styles, or else talk to the LORIS technical team about expanding our Markdown features:
```

Using "e.g." instead of "i.e." because there are many other Markdown features we don't support.

Also rewording the bit about HTML; it doesn't pose a security problem for us as the values are properly escaped. However it might make the text look funny so it's best to avoid it.

File: docs/HelpStyleGuide.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+* `**Bold**`
+* `***Italics and Bold***`
+* `# Heading 1` up to `###### Heading 6`
+  * Use `#H1` for the name of the module at the beginning of the help text
+  * Use `#H2` for any subheadings
+  * Use `#H4` for sub-subheadings
+* Avoid using HTML line break (`<br>`) tags—simply use physical line breaks (spaces between lines) where necessary. 
+  * Using `<br>` might be necessary only in lists of items
+  
+  ## When to Use Emphasis
```
Comment (johnsaigle) [inline docs/HelpStyleGuide.md]:
```suggestion
## When to Use Emphasis
```

I think this needs to be left-aligned in order to render properly as as heading.

File: docs/HelpStyleGuide.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  * i.e. in the Selection Filters section
+  * i.e. in the Metadata column of the table
+  * i.e. in the Imaging menu
+* **Bold** when referring to any clickable button, including modules and tabs
+  * DON’T call it a button
+    * i.e. Click **Submit** (rather than “click the **Submit** button”)
+  * DO call it a module or tab
+    * i.e. In the **Imaging Browser** module
+    * i.e. Click the **Upload** tab
+  * You can also use bold when defining features in a list, to separate the term being defined from the definition itself, i.e. when describing columns in a table:<br>
```
Comment (johnsaigle) [inline docs/HelpStyleGuide.md]:
```suggestion
  * You can also use bold when defining features in a list, to separate the term being defined from the definition itself, e.g. when describing columns in a table:<br>
```

File: docs/HelpStyleGuide.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+Please note, this is NOT about “dumbing things down”. It’s simply about using the most direct language and tone to get the information across to the user in the least amount of time, with the least amount of friction. The quicker a user understands something, the happier they are, the more trust they have with the software, and the less silly questions you can expect to get to the dev mailing list! 
+
+* Use second person point of view rather than third
+  * i.e. “This module allows *you* to browse…” (rather than “This module allows *users* to browse…”)
+* Keep it concise. Avoid unnecessary information.
+  * Try not to be repetitive. 
+  * i.e. the Help text for the Imaging Browser module begins with the title Imaging Browser. You don’t need to start text with “In the Imaging Browser module...”, because the user already knows the module based on the title. Instead, keep it short and say “In this module…”
+* Assume the user is tech-savvy enough in the year 2020 to know about features that are intuitive or common across modern platforms, so you don’t need to repeat in every Help text
```
Comment (johnsaigle) [inline docs/HelpStyleGuide.md]:
```suggestion
* Assume the user is tech-savvy enough to know about features that are intuitive or common across modern platforms, so you don’t need to repeat in every Help text
```

I see why this was added stylistically but I know that we'll forget to keep this current.

File: docs/HelpStyleGuide.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+  * You can also use bold when defining features in a list, to separate the term being defined from the definition itself, e.g. when describing columns in a table:<br>
+     **Column heading 1**: description<br>
+     **Column heading 2**: description<br>
+     Etc.
+     
+## Tone
+
+LORIS is used by scientific community members worldwide, which means it's possible that most users are non-native English speakers. Therefore, it is important that LORIS documentation (setup guides, etc.) be written in a way that is quickly comprehensible by people with many different levels of English language and expression.
+
+This means using more direct language and tone to get information across to the user in the least amount of time, with the least amount of friction. The quicker a user understands something, the happier they are.  Easily understandable documentation also builds trust in the software, in addition to decreasing the amount of support questions we get.
```
Comment (PapillonMcGill) [inline docs/HelpStyleGuide.md]:
```suggestion
This means using more direct language and tone to get information across to the user in the least amount of time, with the least amount of friction. The quicker a user understands something, the happier they are. Easily understandable documentation also builds trust in the software, in addition to decreasing the amount of support questions we get.
```

File: docs/HelpStyleGuide.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+    * i.e. In the **Imaging Browser** module
+    * i.e. Click the **Upload** tab
+  * You can also use bold when defining features in a list, to separate the term being defined from the definition itself, e.g. when describing columns in a table:<br>
+     **Column heading 1**: description<br>
+     **Column heading 2**: description<br>
+     Etc.
+     
+## Tone
+
+LORIS is used by scientific community members worldwide, which means it's possible that most users are non-native English speakers. Therefore, it is important that LORIS documentation (setup guides, etc.) be written in a way that is quickly comprehensible by people with many different levels of English language and expression.
```
Comment (PapillonMcGill) [inline docs/HelpStyleGuide.md]:
```
... LORIS documentation (setup guides, etc.) ...
```
Why `help text` as been eliminated?  An enumeration with only 1 value before `etc`is not proper.

Comment (knoll-alex) [reply docs/HelpStyleGuide.md]: oops! added it back in

**Comment by knoll-alex** [comment]
> Suggested some small changes. Largely looks good and I like the note about plain language!
> 
> In addition to my comments I think there is a bit of an overuse of "i.e." throughout the text and in most cases "e.g." would be more appropriate. It would be nice to fix but I won't block merging this PR because of it.

I've replaced all I.e. with e.g. - thank you for this suggestion. I prefer the way "I.e." looks but sometimes forget it really doesn't mean "for example" !

**Comment by christinerogers** [comment]
Thanks for the last push to get this reviewed, tweaked and merged, guys !
@knoll-alex appreciate the extra edits
@PapillonMcGill @johnsaigle could you re-review?  Let's get this mergeable today. 🔥

**Comment by johnsaigle** [comment]
The image doesn't appear to be rendering but I'm not sure if this is a bug with GitHub. I haven't tried to add in-line images before.

**Comment by christinerogers** [comment]
> The image doesn't appear to be rendering but I'm not sure if this is a bug with GitHub. I haven't tried to add in-line images before.

It renders ok for me in the latest version here: https://github.com/aces/Loris/blob/d3b84eb22ba1ffd39e970aab21a697c394328afe/docs/HelpStyleGuide.md

Which version are you looking at, @johnsaigle ?

File: docs/HelpStyleGuide.md (hunk 11 above)
Comment (christinerogers) [reply docs/HelpStyleGuide.md]: my bad prolly

**Comment by johnsaigle** [comment]
You're right. I was looking at the diff. I forgot GitHub has a two-step process for rendering docs

**Comment by christinerogers** [comment]
@driusan looks pretty ready to merge.

File: docs/HelpStyleGuide.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+## How to Store and Name Files
+
+Help text content is stored into the aces/Loris repo as markdown (.md) files within the help/subdirectory of the module it belongs to. For example, the Help text for Imaging Browser main page is located here: `modules/imaging_browser/help/imaging_browser.md`. 
+
+Some modules have more than one Help text file to correspond with subpages. These files exist in the same `help/` folder, but are named differently - they must match exactly the php/file name that spawns the page. For example, the Imaging Browser module has a subpage called “ViewSession”. Its help file is stored as: `modules/imaging_browser/help/viewsession.md`. 
+
+## Markdown
+
+Formatting should follow markdown guidelines, with some exceptions. This is because LORIS’s current markdown template may not cover all possible markdown features (e.g. bullets). Help text cannot be formatted with HTML so it should not be included in help text. When creating a markdown file, use only the following styles, or else talk to the LORIS technical team about expanding our Markdown features:
```
Comment (driusan) [inline docs/HelpStyleGuide.md]:
```suggestion
Formatting should follow markdown guidelines, with some exceptions. This is because LORIS’s current markdown renderer may not cover all possible markdown features (e.g. bullets). Help text cannot be formatted with HTML so it should not be included in help text. When creating a markdown file, use only the following styles, or else talk to the LORIS technical team about expanding our Markdown features:
```

This is more accurate

File: docs/HelpStyleGuide.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+* `# Heading 1` up to `###### Heading 6`
+  * Use `#H1` for the name of the module at the beginning of the help text
+  * Use `#H2` for any subheadings
+  * Use `#H4` for sub-subheadings
+* Avoid using HTML line break (`<br>`) tags—simply use physical line breaks (spaces between lines) where necessary. 
+  * Using `<br>` might be necessary only in lists of items
+  
+## When to Use Emphasis
+  
+Whether a user knows it or not, they will come to recognize styles that are consistent throughout the text. This way, their eyes can jump to a part of the text that addresses their issue, if they don’t need to read the whole thing. Follow these rules on using emphasis:
```
Comment (driusan) [inline docs/HelpStyleGuide.md]:
```suggestion
Whether a user knows it or not, they will come to recognize styles that are consistent throughout the text. This way, their eyes can jump to a part of the text that addresses their issue, and they don’t need to read the whole thing. Follow these rules on using emphasis:
```

The flow of the sentence is hard to follow because of the "," before "if".. changing it to "and" makes it easier for me to understand (or you could drop the comma, but then it would be a long sentence..)

File: docs/HelpStyleGuide.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+  * e.g. in the Selection Filters section
+  * e.g. in the Metadata column of the table
+  * e.g. in the Imaging menu
+* **Bold** when referring to any clickable button, including modules and tabs
+  * DON’T call it a button
+    * e.g. Click **Submit** (rather than “click the **Submit** button”)
+  * DO call it a module or tab
+    * e.g. In the **Imaging Browser** module
+    * e.g. Click the **Upload** tab
+  * You can also use bold when defining features in a list, to separate the term being defined from the definition itself, e.g. when describing columns in a table:<br>
```
Comment (driusan) [inline docs/HelpStyleGuide.md]:
```suggestion
  * You can also use bold when defining features in a list to separate the term being defined from the definition itself, e.g. when describing columns in a table:<br>
```

I don't think English grammar calls for a comma here.

File: docs/HelpStyleGuide.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+     **Column heading 1**: description<br>
+     **Column heading 2**: description<br>
+     Etc.
+     
+## Tone
+
+LORIS is used by scientific community members worldwide, which means it's possible that most users are non-native English speakers. Therefore, it is important that LORIS documentation (help text, setup guides, etc.) be written in a way that is quickly comprehensible by people with many different levels of English language and expression.
```
Comment (driusan) [inline docs/HelpStyleGuide.md]: English language what and expression?

File: docs/HelpStyleGuide.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+* Keep it concise. Avoid unnecessary information.
+  * Try not to be repetitive. 
+  * e.g. the Help text for the Imaging Browser module begins with the title Imaging Browser. You don’t need to start text with “In the Imaging Browser module...”, because the user already knows the module based on the title. Instead, keep it short and say “In this module…”
+* Assume the user is tech-savvy enough to know about features that are intuitive or common across modern platforms, so you don’t need to repeat in every Help text
+  * e.g. clicking the column headers to re-sort a table
+  * e.g. clicking a Download button to download something
+  * e.g. navigate pages of a table using the arrows provided
+* Use Plain Language
+  * Avoid words that you wouldn't use in everyday speech
+* Where relevant, introduce an abbreviation once and then use that abbreviation for the rest of the copy in that help text (e.g. introduce Quality Control = QC in a help text title, then only call it QC throughout the text)
```
Comment (driusan) [inline docs/HelpStyleGuide.md]:
```suggestion
* Where relevant introduce an abbreviation once and then use that abbreviation for the rest of the copy in that help text (e.g. introduce Quality Control = QC in a help text title, then only call it QC throughout the text)
```

I don't think this comma is necessary either.

File: docs/HelpStyleGuide.md (hunk 15 above)
Comment (johnsaigle) [reply docs/HelpStyleGuide.md]: Maybe it would be more clear to say

```suggestion
LORIS is used by scientific community members worldwide, which means it's possible that most users are non-native English speakers. Therefore, it is important that LORIS documentation (help text, setup guides, etc.) be written in a way that is concise, clear, and free of needlessly ambiguous or complicated words.
```

File: docs/HelpStyleGuide.md (hunk 16 above)
Comment (johnsaigle) [reply docs/HelpStyleGuide.md]: I think the comma is better than no comma. 

Maybe the "Where relevant" can be removed from the beginning of the sentence though. The sentence stands on its own without it.

Or maybe "When introducing an abbreviation, use it once and then...'"

File: docs/HelpStyleGuide.md (hunk 15 above)
Comment (driusan) [reply docs/HelpStyleGuide.md]: I like that better.. I was also a little confused about what users were expressing while reading the help text..

File: docs/HelpStyleGuide.md (hunk 16 above)
Comment (driusan) [reply docs/HelpStyleGuide.md]: I don't feel strongly about this particular comma.

Comment (johnsaigle) [reply docs/HelpStyleGuide.md]: Nor I

**Comment by christinerogers** [comment]
@driusan     see @knoll-alex 's latest changes and re-review? thanks!
